### ISSUE TITLE
Crash occurs when drawing images of special sizes

### ISSUE THREAD

**Issue by FASTSHIFT** [thorvg/thorvg#2391]
Hi, I found that a crash occurs when drawing pictures of special sizes.
It seems that the boundary conditions are not handled properly.

```cpp
int tvg_image_test(void)
{
    Initializer::init(0, CanvasEngine::Sw);

    auto picture = Picture::gen();

    auto buf_size = 31 * 20 * sizeof(uint32_t);
    auto img_buf = (uint32_t*)malloc(buf_size);
    memset(img_buf, 0xAA, buf_size);
    auto res = picture->load(img_buf, 31, 20, true);
    if (res != Result::Success) {
        LV_LOG_ERROR("Failed to load the image");
        return -1;
    }

    // picture->scale(0.2);

    float w, h;
    res = picture->size(&w, &h);
    if (res != Result::Success) {
        LV_LOG_ERROR("Failed to get the image size");
        return -1;
    }

    LV_LOG_USER("Image size: %f x %f", w, h);

    auto draw_buf = lv_draw_buf_create(w, h, LV_COLOR_FORMAT_ARGB8888, LV_STRIDE_AUTO);

    if (!draw_buf) {
        LV_LOG_ERROR("Failed to create draw buffer");
        return -1;
    }

    Matrix matrix;
    matrix.e11 = 0.707106769;
    matrix.e12 = -0.707106769;
    matrix.e13 = -3.53553391;
    matrix.e21 = 0.707106769;
    matrix.e22 = 0.707106769;
    matrix.e23 = 3.53553391;
    matrix.e31 = 0.0;
    matrix.e32 = 0.0;
    matrix.e33 = 1.0;
    picture->transform(matrix);

    auto shape = Shape::gen();
    shape->appendRect(0, 0, 31, 20, 0, 0);
    shape->transform(matrix);
    picture->composite(std::move(shape), CompositeMethod::ClipPath);

    auto canvas = SwCanvas::gen();
    canvas->target((uint32_t*)draw_buf->data, w, w, h, SwCanvas::ARGB8888);
    canvas->push(std::move(picture));

    canvas->draw();
    canvas->sync();
    canvas->clear(true, true);

    lv_obj_t* img = lv_image_create(lv_scr_act());
    lv_image_set_src(img, (const lv_img_dsc_t*)draw_buf);
    return 0;
}
```

![image](https://github.com/thorvg/thorvg/assets/26767803/98bc1e1d-55a8-47bf-8e6e-8acacc59262c)

**Comment by hermet** [comment]
@FASTSHIFT we will check it soon, thanks.

**Comment by FASTSHIFT** [comment]
> @FASTSHIFT we will check it soon, thanks.

Thank you so much! I added [VG-Lite simulator test](https://github.com/lvgl/lvgl/pull/6264) to LVGL so that ThorVG can also participate in the test, This Crash was found [here](https://github.com/lvgl/lvgl/actions/runs/9463350903/job/26068332178?pr=4883#step:7:4160).

**Comment by hermet** [comment]
@FASTSHIFT the fix will be in v0.13.8. thanks.

**Comment by FASTSHIFT** [comment]
> @FASTSHIFT the fix will be in v0.13.8. thanks.

It works well, thanks for your help!
